rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/TaskController.java

```diff
@@ -544,17 +549,75 @@ public ResponseEntity<TaskRepresentationModel> terminateTask(@PathVariable Strin
+          TransferTaskRepresentationModel transferTaskRepresentationModel)
       throws TaskNotFoundException,
           WorkbasketNotFoundException,
           NotAuthorizedOnWorkbasketException,
           InvalidTaskStateException {
-    Task updatedTask =
-        taskService.transfer(taskId, workbasketId, setTransferFlag == null || setTransferFlag);
+    Task updatedTask;
+    if (transferTaskRepresentationModel == null) {
+      updatedTask = taskService.transfer(taskId, workbasketId);
+    } else {
+      updatedTask =
+          taskService.transferWithOwner(
+              taskId,
+              workbasketId,
+              transferTaskRepresentationModel.getOwner(),
+              transferTaskRepresentationModel.getSetTransferFlag());
+    }
 
     return ResponseEntity.ok(taskRepresentationModelAssembler.toModel(updatedTask));
   }
 
+  /**
+   * This endpoint transfers a list of Tasks listed in the body to a given Workbasket, if possible.
+   * Tasks that can be transfered without throwing an exception get transferred independent of other
+   * Tasks. If the transfer of a Task throws an exception, then the Task will remain in the old
+   * Workbasket.
+   *
+   * @title Transfer Tasks to another Workbasket
+   * @param workbasketId the Id of the destination Workbasket
+   * @param transferTaskRepresentationModel JSON formatted request body containing the TaskIds,
+   *     owner and setTransferFlag of tasks to be transferred; owner and setTransferFlag are
+   *     optional, while the TaskIds are mandatory
+   * @return the successfully transferred Task.
+   * @throws WorkbasketNotFoundException if the requested Workbasket does not exist
+   * @throws NotAuthorizedOnWorkbasketException if the current user has no authorization to transfer
+   *     the Task
+   */
+  @PostMapping(path = RestEndpoints.URL_TRANSFER_WORKBASKET_ID)
+  @Transactional(rollbackFor = Exception.class)
+  public ResponseEntity<Map<String, Object>> transferTasks(
+      @PathVariable String workbasketId,
+      @RequestBody TransferTaskRepresentationModel transferTaskRepresentationModel)
+      throws NotAuthorizedOnWorkbasketException, WorkbasketNotFoundException {
+    List<String> taskIds = transferTaskRepresentationModel.getTaskIds();
+    BulkOperationResults<String, TaskanaException> result;
+    result =
+        taskService.transferTasksWithOwner(
+            workbasketId,
+            taskIds,
+            transferTaskRepresentationModel.getOwner(),
+            transferTaskRepresentationModel.getSetTransferFlag());
+
+    Set<String> failedIds = new HashSet<>(result.getFailedIds());
+    List<String> successfullyTransferredTaskIds =
+        taskIds.stream().filter(not(failedIds::contains)).toList();
+
+    Map<String, ErrorCode> newErrorMap = new HashMap<>();
+
+    for (Map.Entry<String, TaskanaException> entry : result.getErrorMap().entrySet()) {
+      newErrorMap.put(entry.getKey(), entry.getValue().getErrorCode());
```

Comment:
Why do you use the error code instead of the value?

Reply:
The Value contains the stack trace of the error, which will result in the body being too long